lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -434,88 +437,207 @@ describe('BasicLTILaunchApp', () => {
+      () => {
+        beforeEach(() => {
+          fakeConfig.canvas.speedGrader = {
+            submissionParams: {
+              lis_result_sourcedid: 'modelstudent-assignment1',
+            },
+          };
+          fakeConfig.viaUrl = 'https://via.hypothes.is/123';
+          // The actual value of this configuration property in practice would
+          // be an object, but that's unimportant to this component, which just
+          // checks for its truthy presence.
+          fakeConfig.hypothesisClient.reportActivity = true;
+        });
 
-    it('does not report the submission when there is no `viaUrl` provided', async () => {
-      // When present, viaUrl becomes the contentUrl
-      fakeConfig.viaUrl = null;
-      renderLTILaunchApp();
-      assert.isTrue(fakeApiCall.notCalled);
-    });
+        it('does not report the submission when the content iframe starts loading', async () => {
+          const wrapper = renderLTILaunchApp();
+          await new Promise(resolve => setTimeout(resolve, 1));
+          assert.notCalled(fakeApiCall);
 
-    it('reports a submission if VitalSource configuration is present, even if `viaUrl` not set', async () => {
-      fakeConfig.viaUrl = null;
-      // Provide a VitalSource configuration object as an alternative
-      fakeConfig.vitalSource = { foo: 'bar' };
-      renderLTILaunchApp();
+          // After the successful API call, the iframe should still be rendered.
+          wrapper.update();
+          assert.isTrue(wrapper.exists('iframe'));
+        });
 
-      assert.calledWith(fakeApiCall, {
-        authToken: 'dummyAuthToken',
-        path: '/api/lti/submissions',
-        data: fakeConfig.canvas.speedGrader.submissionParams,
-      });
-    });
+        it('registers a callback for `onAnnotationActivity`', async () => {
+          renderLTILaunchApp();
+
+          const onCall = fakeRpcServer.on.getCall(0);
+          assert.equal(onCall.args[0], 'annotationActivity');
+          assert.isFunction(onCall.args[1]);
+        });
+
+        it('submits a submission when `onAnnotationActivity` callback invoked', async () => {
+          renderLTILaunchApp();
+
+          const callback = fakeRpcServer.on.getCall(0).args[1];
+          assert.notCalled(fakeApiCall);
+          callback('create', {});
+          await new Promise(resolve => setTimeout(resolve, 1));
+
+          assert.calledOnce(fakeApiCall);
+
+          const apiCall = fakeApiCall.getCall(0);
+          assert.deepEqual(apiCall.args[0], {
+            authToken: 'dummyAuthToken',
+            path: '/api/lti/submissions',
+            data: {
+              submitted_at: undefined,
+              ...fakeConfig.canvas.speedGrader.submissionParams,
+            },
+          });
+        });
+
+        it('deregisters callback after submission is made', async () => {
+          renderLTILaunchApp();
+
+          const callback = fakeRpcServer.on.getCall(0).args[1];
```

Comment:
This test will break if another callback gets added for some other RPC event. It might be useful to create a helper that finds the call on a spy that matches a particular set of arguments (eg. for uses cases like finding the API call where the `path` parameter has this value, or finding the event emitter `on` call where the event name has a certain value).